services/spar/src/Spar/Scim/User.hs

```diff
@@ -243,32 +253,38 @@ createValidScimUser
 createValidScimUser (ValidScimUser user uref handl mbName richInfo) = do
     -- Generate a UserId will be used both for scim user in spar and for brig.
     buid <- Id <$> liftIO UUID.nextRandom
-
     -- ensure uniqueness constraints of all affected identifiers.
+    -- if we crash now, retry POST will just work
     assertUserRefUnused uref
     assertHandleUnused handl buid
-
-    -- Create SCIM user here in spar.
+    -- if we crash now, retry POST will just work, or user gets told the handle
+    -- is already in use and stops POSTing
+   
+    -- TODO(arianvp): Get rid of manual lifting. Needs to be SCIM instances for ExceptT
+    -- This is the pain and the price you pay for the horribleness called MTL
     storedUser <- lift $ toScimStoredUser buid user
-    lift . wrapMonadClient $ Data.insertScimUser buid storedUser
-
-    -- Create SAML user here in spar, which in turn creates a brig user. The user is created
-    -- with 'ManagedByScim', which signals to client apps that the user should not be editable
-    -- from the app (and ideally brig should also enforce this). See {#DevScimOneWaySync}.
-    lift $ createSamlUserWithId buid uref mbName ManagedByScim
-
-    -- Set user handle on brig (which can't be done during user creation yet).
-    -- TODO: handle errors better here?
+    idpConfig <-  lift $ SAML.getIdPConfigByIssuer (uref ^. SAML.uidTenant)
+    let teamid = view SAML.idpExtraInfo idpConfig
+    buid' <- lift $ Intra.Brig.createBrigUser uref buid teamid mbName ManagedByScim
+    assert (buid == buid') $ pure ()
+    -- If we crash now, we have an active user that cannot login. And can not
+    -- be bound this will be a zombie user that needs to be manually cleaned
+    -- up.  We should consider making setUserHandle part of createUser and
+    -- making it transactional.  If the user redoes the POST A new standalone
+    -- user will be created
     lift $ Intra.Brig.setBrigUserHandle buid handl
-
-    -- Set rich info on brig
+    -- If we crash now,  a POST retry will fail with 409 user already exists. 
+    -- Azure at some point will retry with GET /Users?filter=userName eq handle
+    -- and then issue a PATCH containing the rich info and the externalId
+    -- TODO(arianvp): Implement PATCH (Part of Phase 1)
     lift $ Intra.Brig.setBrigUserRichInfo buid richInfo
-
+    -- If we crash now, same as above, but the PATCH will only contain externalId
+    
+    -- TODO(arianvp): these two actions we probably want to make transactional
```

Comment:
Right, I think this might still be worth it in this one, and should look like something in the lines of:
```
insertScimAndSamlUser :: (HasCallStack, MonadClient m) => UserId -> SAML.UserRef -> ScimC.User.StoredUser SparTag -> m ()
insertScimAndSamlUser uid (SAML.UserRef tenant subject) scimUsr = retry x5 . batch $ do
  setType BatchLogged
  setConsistency Quorum
  addPrepQuery insScim (uid, WrappedScimStoredUser scimUsr)
  addPrepQuery insSaml (tenant, subject, uid)
  where
    insScim :: PrepQuery W (UserId, WrappedScimStoredUser SparTag) ()
    insScim = "INSERT INTO scim_user (id, json) VALUES (?, ?)"

    insSaml :: PrepQuery W (SAML.Issuer, SAML.NameID, UserId) ()
    insSaml = "INSERT INTO user (issuer, sso_id, uid) VALUES (?, ?, ?)"
```

The main annoyance I see with this is that you will need a `Show` instance for `WrappedScimStoredUser SparTag` -> not sure how trivial it is to come up with a meaningful one.